utils/src/com/cloud/utils/StringUtils.java

```diff
@@ -32,6 +33,21 @@
 public class StringUtils {
     private static final char[] hexChar = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
 
+    private static Charset _ACS_Charset;
+
+    {
+        String preferredCharset = "UTF8";
```

Comment:
so are so absolutely right. My code would silently fall back to that as it is the default if file.encoding is not set. bad bug, will fix.